lib/Macros/Sources/ObservationMacros/Availability.swift

```diff
@@ -0,0 +1,148 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+import SwiftSyntaxMacros
+import SwiftDiagnostics
+import SwiftOperators
+import SwiftSyntaxBuilder
+
+
+extension AttributeSyntax {
+  var availability: AttributeSyntax? {
+    if attributeName.identifier == "available" {
+      return self
+    } else {
+      return nil
+    }
+  }
+}
+
+extension IfConfigClauseSyntax.Elements {
+  var availability: IfConfigClauseSyntax.Elements? {
+    switch self {
+    case .attributes(let attributes):
+      if let availability = attributes.availability {
+        return .attributes(availability)
+      } else {
+        return nil
+      }
+    default:
+      return nil
+    }
+  }
+}
+
+extension IfConfigClauseSyntax {
+  var availability: IfConfigClauseSyntax? {
+    if let availability = elements?.availability {
+      return IfConfigClauseSyntax(
+        leadingTrivia: leadingTrivia,
+        unexpectedBeforePoundKeyword,
+        poundKeyword: poundKeyword,
+        unexpectedBetweenPoundKeywordAndCondition,
+        condition: condition,
+        unexpectedBetweenConditionAndElements,
+        elements: availability,
+        unexpectedAfterElements,
+        trailingTrivia: trailingTrivia
+      )
+    } else {
+      return nil
+    }
+  }
+  
+  var clonedAsIf: IfConfigClauseSyntax {
+    IfConfigClauseSyntax(
+      leadingTrivia: leadingTrivia,
+      unexpectedBeforePoundKeyword,
+      poundKeyword: .poundIfKeyword(),
```

Comment:
The intent here seems to be to get a clone; I think you can return `self.detached` instead of duplicating everything

Reply:
so this isn't per-se a clone. It is a near miss, because it transforms `#elseif` statements that are the first element in the branches (because of trimming of attributes) to just `#if` statements.

e.g.

```
#if FOO
@someOtherAttributeOtherThanAvailability
#elseif BAR
@available(...)
#endif
```

into 
```
#if BAR
@available(...)
#endif
```

Reply:
unless do you mean this: `detached.with(\.poundKeyword, .poundIfKeyword())`?

Reply:
that worked